authorization/role/service/role_management_service.go

```diff
@@ -95,3 +109,29 @@ func (r *RoleManagementServiceImpl) ListByResourceAndRoleName(ctx context.Contex
 
 	return roles, err
 }
+
+// Assign assigns a user/identity to role, for a specific resource
+func (r *RoleManagementServiceImpl) Assign(ctx context.Context, identityIDs []uuid.UUID, resourceID string, roleName string) error {
+
+	// TODO: Check if the user is authorized to assign roles to other users.
```

Comment:
yeah, I could move out the `check for scope` as a service method, and then consume it in the controller.

Reply:
>  why does this service class contain code that accesses the repositories directly?

@sbryzak , because things like checking if resource exists and then reporting the right error felt more like an operation closer to the business logic. But I see your point.

Reply:
Now that the 'service' layer is gone, the business logic in the controller would include:

- check if the user X who is adding user Y to the space has a role with the `assign_role` scope. If not return, 403
- check if the user already belongs to the space. if not, return 403